waku/v2/node/localnode.go

```diff
@@ -228,8 +234,28 @@ func selectCircuitRelayListenAddresses(addresses []ma.Multiaddr) ([]ma.Multiaddr
 	return result, nil
 }
 
-func (w *WakuNode) getENRAddresses(addrs []ma.Multiaddr) (extAddr *net.TCPAddr, multiaddr []ma.Multiaddr, err error) {
+func filter0Port(addresses []ma.Multiaddr) ([]ma.Multiaddr, error) {
+	var result []ma.Multiaddr
+	for _, addr := range addresses {
+		portStr, err := addr.ValueForProtocol(ma.P_TCP)
+		if err != nil && !errors.Is(err, multiaddr.ErrProtocolNotFound) {
+			return nil, err
+		}
 
+		port, err := strconv.Atoi(portStr)
+		if err != nil {
+			return nil, err
+		}
+
+		if port != 0 {
+			result = append(result, addr)
+		}
+	}
+
+	return result, nil
+}
+
+func (w *WakuNode) getENRAddresses(addrs []ma.Multiaddr) (extAddr *net.TCPAddr, multiaddr []ma.Multiaddr, err error) {
 	extAddr, err = selectMostExternalAddress(addrs)
```

Comment:
Also i was going through waku ENR spec and noticed that practically we can include upto 3 multiaddresses. Maybe we should try to include other addresses as well in that case rather than only the single most external address?

https://github.com/waku-org/specs/blob/master/standards/core/enr.md#multiaddrs-enr-key

Reply:
We try to include as many multiaddresses as possible: https://github.com/waku-org/go-waku/blob/795322a19665ac48ca1f5cf725f64b1fd1e67808/waku/v2/protocol/enr/localnode.go#L33-L54

The most external address is used only to populate the `tcp` and `ip` keys:
> For raw TCP & UDP connections details, [EIP-778](https://eips.ethereum.org/EIPS/eip-778) pre-defined keys SHOULD be used; The keys tcp, udp, ip (and tcp6, udp6, ip6 for IPv6) are enough to convey all necessary information;

Reply:
Hmm, while i was testing locally..i had noticed that my LAN address was always getting added to ENR eventhough i have circuit-relay address. 
I am wondering why that could be happening when there are no other external addresses.

Reply:
Realized after our discussions that Waku multiaddrs key is not decoded by external tools. They seem to be deriving multiaddr based on other fields.